File: ContractionTimer/src/android/java/com/ianhanniballake/contractiontimer/backup/ContractionTimerBackupAgent.java
package com.ianhanniballake.contractiontimer.backup;

import android.annotation.TargetApi;
import android.app.backup.BackupAgentHelper;
import android.app.backup.SharedPreferencesBackupHelper;

/**
 * Agent which handles com.ianhanniballake.contractiontimer.backup of user settings
 */
@TargetApi(8)
public class ContractionTimerBackupAgent extends BackupAgentHelper {
    /**
     * Gets the name associated with PreferenceManager.getDefaultPreferences(), as given by the Android source code
     *
     * @return The name associated with PreferenceManager.getDefaultPreferences()
     */
    public String getDefaultSharedPreferencesName() {
        return getPackageName() + "_preferences";
    }

    @Override
    public void onCreate() {
        final SharedPreferencesBackupHelper helper = new SharedPreferencesBackupHelper(this,
                getDefaultSharedPreferencesName());
        addHelper("preferences", helper);
    }
}

File: mobile/src/main/java/com/ianhanniballake/contractiontimer/ContractionTimerBackupAgent.kt
package com.ianhanniballake.contractiontimer

import android.app.backup.BackupAgentHelper
import android.app.backup.SharedPreferencesBackupHelper

/**
 * Agent which handles com.ianhanniballake.contractiontimer backup of user settings
 */
class ContractionTimerBackupAgent : BackupAgentHelper() {
    /**
     * Gets the name associated with PreferenceManager.getDefaultPreferences(), as given by the Android source code
     *
     * @return The name associated with PreferenceManager.getDefaultPreferences()
     */
    private val defaultSharedPreferencesName: String
        get() = packageName + "_preferences"

    override fun onCreate() {
        val helper = SharedPreferencesBackupHelper(this, defaultSharedPreferencesName)
        addHelper("preferences", helper)
    }
}
